### PR TITLE
Don't take collateral if validation doesn't fail in phase 2

### PR DESCRIPTION
**Pull request by MaximilianAlgehed**
Pre-submit checklist:
- Branch
    - [ ] Tests are provided (if possible)
    - [ ] Commit sequence broadly makes sense
    - [ ] Key commits have useful messages
    - [ ] Formatting, PNG optimization, etc. are updated
- PR
    - [ ] Self-reviewed the diff
    - [ ] Useful pull request description
    - [ ] Reviewer requested

### CODE REVIEW

**Comment by MaximilianAlgehed** [comment]
@sjoerdvisscher here is a PR with that fix. I think we might need a bit of help debugging the PAB issue. Should we make an issue with this too?

File: plutus-ledger/src/Ledger/Index.hs [hunk 1]
```diff
@@ -178,8 +178,7 @@ validateTransactionOffChain t = do
 
         idx <- vctxIndex <$> ask
         pure (Nothing, insert t idx)
-        )
-    `catchError` payCollateral
+        ) `catchError` payCollateral
```
Comment (koslambrou) [inline plutus-ledger/src/Ledger/Index.hs:181]: I don't understand how this actually changed the result of PAB testing? Isn't this a code format change?

Comment (koslambrou) [reply plutus-ledger/src/Ledger/Index.hs:181]: Oh, just saw your comment on the other PR: "because of how do and infix notation interact"

Comment (koslambrou) [reply plutus-ledger/src/Ledger/Index.hs:181]: But I guess I don't see the difference...

Comment (MaximilianAlgehed) [reply plutus-ledger/src/Ledger/Index.hs:181]: This is super subtle, the old code would do `catchError` on the whole do-block, not just the bits in the brackets (the phase 2 validation).

**Comment by sjoerdvisscher** [comment]
I have looked at this a bit. The problem seems to be that the test was not ok to begin with. The 3rd test does not fail if you comment out the second test. So there's some state transfer between tests. @koslambrou it seems you wrote these tests, does that sound plausible?

**Comment by koslambrou** [comment]
Found the error, but I don't know why your fix triggered it.

It's in the test case "PingPong contract state is maintained across PAB instances" in `CliSpec`.

Change

```
let newConfig = bumpConfig 50 "db1" pabConfig
```

to

```
 let newConfig = bumpConfig 50 "db1" defaultPabConfig
```

That solved it locally.

**Comment by UlfNorell** [comment]
@koslambrou if the test failure is unrelated to this fix, maybe it makes more sense to fix the test in a separate PR and hold this PR until that has landed? If you want it here you can always make a PR to this branch.

**Comment by koslambrou** [comment]
@MaximilianAlgehed The fix is on `main`. You can rebase.

**Comment by UlfNorell** [comment]
@koslambrou Rebased and tests are passing.

**Pull request merged**